x-pack/plugin/runtime-fields/qa/with-security/src/test/java/org/elasticsearch/xpack/security/PermissionsIT.java

```diff
@@ -0,0 +1,244 @@
+            "}\n");
+        assertOK(adminClient().performRequest(indexDoc1));
+
+        Request indexDoc2 = new Request("PUT", "/fls/_doc/2");
+        indexDoc2.setJsonEntity("{\n" +
+            "    \"hidden\" : \"should not be read\"\n" +
+            "}\n");
+        assertOK(adminClient().performRequest(indexDoc2));
+
+        Request indexDoc3 = new Request("PUT", "/fls/_doc/3");
+        indexDoc3.addParameter("refresh", "wait_for");
+        indexDoc3.setJsonEntity("{\n" +
+            "    \"hidden\" : \"should not be read\"\n" +
+            "}\n");
+        assertOK(adminClient().performRequest(indexDoc3));
+
+        SearchRequest searchRequest = new SearchRequest("fls").source(new SearchSourceBuilder().docValueField("hidden_values_count"));
+        {
+            SearchResponse searchResponse = adminHighLevelClient.search(searchRequest, RequestOptions.DEFAULT);
+            assertEquals(3, searchResponse.getHits().getTotalHits().value);
+            for (SearchHit hit : searchResponse.getHits().getHits()) {
+                assertEquals(1, hit.getFields().size());
+                assertEquals(1, (int)hit.getFields().get("hidden_values_count").getValue());
+            }
+        }
+        {
+            SearchResponse searchResponse = highLevelClient.search(searchRequest, RequestOptions.DEFAULT);
+            assertEquals(3, searchResponse.getHits().getTotalHits().value);
+            for (SearchHit hit : searchResponse.getHits().getHits()) {
+                assertEquals(0, (int)hit.getFields().get("hidden_values_count").getValue());
+            }
+        }
+    }
+
+    public void testFLSOnRuntimeField() throws IOException {
+        Request createIndex = new Request("PUT", "/fls");
+        createIndex.setJsonEntity("{\n" +
+            "    \"mappings\" : {\n" +
+            "        \"properties\" : {\n" +
+            "            \"date\" : {\"type\" : \"keyword\"},\n" +
+            "            \"year\" : {\n" +
+            "              \"type\" : \"runtime_script\", \n" +
+            "              \"runtime_type\" : \"keyword\",\n" +
+            "              \"script\" : \"emitValue(doc[\\\"date\\\"].value.substring(0,4))\"\n" +
+            "            }\n" +
+            "        }\n" +
+            "    }\n" +
+            "}\n");
+        assertOK(adminClient().performRequest(createIndex));
+
+        Request indexDoc1 = new Request("PUT", "/fls/_doc/1");
+        indexDoc1.setJsonEntity("{\n" +
+            "    \"date\" : \"2009-11-15T14:12:12\"\n" +
+            "}\n");
+        assertOK(adminClient().performRequest(indexDoc1));
+
+        Request indexDoc2 = new Request("PUT", "/fls/_doc/2");
+        indexDoc2.setJsonEntity("{\n" +
+            "    \"date\" : \"2016-11-15T14:12:12\"\n" +
+            "}\n");
+        assertOK(adminClient().performRequest(indexDoc2));
+
+        Request indexDoc3 = new Request("PUT", "/fls/_doc/3");
+        indexDoc3.addParameter("refresh", "wait_for");
```

Comment:
I don't think we need to wait_for a refresh, we can just do `true`, I think.

Reply:
both work I guess